marshal/consumer_test.go

```diff
@@ -306,6 +306,27 @@ func (s *ConsumerSuite) TestMultiTopicClaim(c *C) {
 	c.Assert(cn.getNumActiveClaims(), Equals, partitions)
 	c.Assert(len(claimedTopics), Equals, len(topics))
 	wg.Wait()
+
+	// let's force another claimTopics and make sure we don't get any more notifications.
+	// Force our consumer to run it's topic claim loop so we know it has run
+	cn.claimTopics()
+	timeoutChan := make(chan struct{})
+	go func() {
+		select {
+		case claimedTopics := <-cn.TopicClaims():
+			// we should not get any more topic claims
+			log.Errorf("received topic claims unexpectedly: %v\n", claimedTopics)
+			break
+		case <-time.After(time.Second):
```

Comment:
I think it's sort of odd to call one of the core features of the language (channel reads & writes) confusing but I don't care enough to argue further :)